src/Microsoft.Restier.Providers.EntityFramework/Model/ModelProducer.cs

```diff
@@ -62,7 +62,19 @@ public Task<IEdmModel> GetModelAsync(ModelContext context, CancellationToken can
             var dbContext = context.GetApiService<DbContext>();
 
             var efModel = (dbContext as IObjectContextAdapter).ObjectContext.MetadataWorkspace;
-            var efEntityContainer = efModel.GetItems<EntityContainer>(DataSpace.CSpace).Single();
+            var efEntityContainers = efModel.GetItems<EntityContainer>(DataSpace.CSpace);
+            var efEntityContainer = efEntityContainers.FirstOrDefault(c => c.Name == dbContext.GetType().Name);
+            if (efEntityContainer == null)
+            {
+                if (efEntityContainers.Count > 1)
+                {
```

Comment:
Do we know which case we will have two container, what will each container will have? If we fix like this, and EF do have two containers, each container will have some entity set (DbSet), then the logic will be wrong, we will need to iterate the container to get the information.

I am OK to add this as a temp fix but please help to add a TODO comment here for further improvement.